demos/smart_classroom_demo/src/action_detector.cpp

```diff
@@ -69,15 +70,64 @@ ActionDetection::ActionDetection(const ActionDetectorConfig& config)
         inputInfoFirst->setPrecision(Precision::U8);
         inputInfoFirst->getInputData()->setLayout(Layout::NCHW);
 
+        const auto input_dims = inputInfoFirst->getDims();
+        network_input_size_.height = input_dims[1];
+        network_input_size_.width = input_dims[0];
+
         OutputsDataMap outputInfo(net_reader.getNetwork().getOutputsInfo());
 
         for (auto&& item : outputInfo) {
             item.second->setPrecision(Precision::FP32);
             item.second->setLayout(InferenceEngine::TensorDesc::getLayoutByDims(item.second->getDims()));
         }
 
+        new_network_ = outputInfo.find(config_.new_loc_blob_name) != outputInfo.end();
```

Comment:
I think that proposed naming style is unsuccessful because we have many old and in future new models. So, the prefix `new_` is a transition point between families of models.

Reply:
There is nothing said about old and new families (at least in this repo). For a reader who is unaware of this split this may seem strange why in some demos some outputs (and other vars) are refereed as old/new.

I mean if old/new split will appear only in demo var naming this will be unclear for understanding. It will be harder to read the code because it will require matching new/old var names with model properties in mind.

In case of future model naming with new/old words I think it may be OK for vars to have the same prefixes, but this is not that case.

Reply:
I agree with Vladimir. In fact, I would suggest to stick the settings relevant to only one variant of a model into a struct (say, `Pdar0005Config` and `Pdar0006Config`, to avoid prefixing each and every one of them.

Reply:
So, what do you propose in this case (we cannot use 0005 and 0006 names)? In OMZ we have have 3 old-style models and 1 new-style model and demo should work with all of them (note that new model is not a replacement of the old one).

Reply:
Oh, I see. Fixed.

Reply:
What are the names of these 3 old-style models and 1 new-style model. it is not clear for me.

The description (smart_classroom_demo/README.md) says only about `person-detection-action-recognition-0005` and `person-detection-action-recognition-0006`.

Reply:
If you will continue explore README file you see the list of models: `person-detection-action-recognition-0005, person-detection-action-recognition-0006, person-detection-raisinghand-recognition-0001, person-detection-action-recognition-teacher-0002`...

Reply:
I see now, thank you.

I like Roman's idea about config structs. We can introduce a config for each model (but it will introduce duplication in values of some fields).

What about a config struct that accepts output info in constructor and initializes its fields (`anchor_height`, `anchor_width`, `loc_blob_name` and so on (may be this fields should be immutable)) accordingly?

Reply:
I cannot understand what is the reason to speak so many words about parameter which the user will never set manually because the demo application configures self according the structure of input model automatically and such parameter prefix (old/new) is only to store them in one logical place...

Reply:
I believe code readability alongside with good design are important properties, especially for samples and demos which are referenced as how to use OpenVINO. The part we are discussing is not about IE API, but it still affects overall code quality.

> parameter prefix (old/new) is only to store them in one logical place

 \- there are four places now: `ActionDetectorConfig`, `ActionDetection::GetDetections`, `ActionDetection::fetchResults()` and `ActionDetection::ActionDetection`. I was hoping that we could at least move that naming mess into a separate structure's constructor.

Reply:
We have different pipelines to process old/new style networks and it's not enough to rename/split config or what ever. So, in another words, you prose to refactor the whole code of demo (or split it in different demo - it's impossible for our customer) even it should be merged ASAP. Moreover, this demo is not to demonstrate any ability of OpenVINO. It's to support very specific models in very specific domain.

Reply:
OK. I agree with the statement about different postprocessing.